src/homework4.py

```diff
@@ -0,0 +1,421 @@
+import sys
+
+import numpy as np
+import pandas as pd
+import statsmodels.api as sm
+from plotly import express as px
+from plotly import figure_factory as ff
+from plotly import graph_objects as go
+from plotly.subplots import make_subplots
+from sklearn.ensemble import RandomForestClassifier
+
+
+def response_type_check(df, response):
+    # Determine if response is continuous or boolean
+    unique_responses = df[response].nunique()
+
+    if unique_responses == 2:
+        boolean_check = True
+        print("Response is boolean\n")
+    else:
+        boolean_check = False
+        print("Response is continuous\n")
+
+    return boolean_check
+
+
+def cat_cont_check(df, predictor):
+    # Determine if the predictor is cat/cont
+    if isinstance(df[predictor][0], float):
+        cat_check = False
+        print("Predictor {} is continuous\n".format(predictor))
```

Comment:
the prof gave me an advice earlier in the semester for hw1, that you can use print(f"Predictor {predictor} is continuous\n".) instead of the .format at the end, that's slightly less code. However, totally depending on personal preference, your code works absolutely fine how it is!